Make May 2026 news public (#33018)
idaholab/moose: modules/doc/content/newsletter/2026/2026_05.md

### Subchannel module

- The repeated laminar/transitional/turbulent Nusselt-number blending used by the heat transfer
  coefficient closures was centralized into `SCMHTCClosureBase::blendTurbulentNusseltNumber()`, and
  new mixing closures were added: [SCMMixingChengTodreas.md], [SCMMixingConstantBeta.md], and
  [SCMMixingKimAndChung.md], the latter rejecting wire-wrapped assemblies. Validity diagnostics were
  added for the mixing closures.
- Subchannel auxiliary variables can now be added directly, the geometry initial conditions are
  auto-populated, and the hydraulic geometry calculations were moved to the mesh, with regression tests
  added for the deformation geometry.

### Fluid Properties module

The [HeliumFluidProperties.md] gained AD support for computing the specific enthalpy `h` from pressure
and temperature, along with its derivatives.

### Navier Stokes module

The weakly-compressible Navier-Stokes Physics now automatically set the time derivatives of constant
variables to zero, and a warning that was unfairly raised when using `system_names` and user objects
for the pressure constraint was silenced.

### Porous Flow module

> A bug in the `PorousFlowUnsaturated` action was fixed.

Reviewer: seems claude isnt good at it

Reviewer: I ll do a round of clean up when I have time, maybe tomorrow

  variable on an $H(\mathrm{curl})$ Nedelec space into a target variable on an $H(\mathrm{div})$
  Raviart-Thomas space. In 2D the two bases are related by a 90-degree rotation, which is useful in
  magnetostatic postprocessing for obtaining the magnetic flux density from a rotated gradient.
  ([idaholab/moose#32923](https://github.com/idaholab/moose/pull/32923))
- Several cleanups were made to the MFEM functor materials and transfers, and the MFEM documentation
  was simplified by removing trailing whitespace and unifying the form of internal links.
  ([idaholab/moose#32781](https://github.com/idaholab/moose/pull/32781),
  [idaholab/moose#32884](https://github.com/idaholab/moose/pull/32884))

### Kokkos Backend Improvements

- Automatic differentiation (AD) support was added to Kokkos-MOOSE. The residual object registration
  macros were unified, the `datum.reinit()` call was removed, and jagged array data is now padded by
  one to avoid reading past the end of the data.
- Kokkos reducers received a revised `init`/`join` signature and improved timing accuracy.
- Kokkos materials are now only added to the master thread warehouse, and material block restriction

>   setup was moved from `initialSetup` to a mesh initialization hook.

Reviewer: @NamjaeChoi thoughts about this? good enough?

Reviewer 2: Is it AI generated?

Reviewer: Yeah

Reviewer 2: Will do some revision.

Reviewer: thanks. i ll keep in mind you're giving examples on how to do this and get claude to do that next time

### Array variable value volume histogram

The [ArrayVariableValueVolumeHistogram.md] vector postprocessor was added to tabulate, for each
component of an array variable, the volumes of the domain in which the variable holds values within
a set of bins. The bin range and count are controlled with the `min_value`, `max_value`, and
`bin_number` parameters.

### Manifold-based subdomain assignment

The [ManifoldSubdomainGenerator.md] was added to assign a subdomain ID based on whether an element's
vertex-average point lies inside a closed (watertight) surface mesh, supplied through the mesh
generator pipeline. Beyond tagging elements inside or outside an arbitrary surface, it can be used to
"voxelize" a geometry defined by a surface mesh by removing elements of a structured grid that fall

> inside or outside the manifold, which is useful when a tetrahedral mesh is not desired.

Reviewer: ```suggestion
inside or outside the manifold, which is useful when exactly fitting to the boundary is not required.
```

Reviewer: @zachmprince is this good enough?

### Advanced extrusion along a curve and radial growth

The [AdvancedExtruderGenerator.md] can now extrude a mesh following a line mesh / curve rather than a
fixed direction, with the extrusion direction computed locally at every curve node. The starting and
ending directions can optionally be imposed. In addition, the radial extent of the extruded mesh can
be progressively expanded or diminished throughout the extrusion by setting the `end_radial_extent`
parameter.

### Polygon mesh extrusion in parallel

The conversion of polygon meshes to tetrahedra and the extrusion of polygon meshes now work with
distributed meshes, with unique ID and midnode numbering made consistent in parallel. The optional

> PNG output was also made more robust, skipping output on a libpng ABI mismatch instead of segfaulting.

Reviewer: ```suggestion
PNG output was also made more robust and additional parameters were added to better select the area to plot.
```

### Junction component

The [ComponentJunction.md] action component was added to create a contiguous junction between two
components, either by stitching meshes that are already in contact or by extruding one boundary
towards the other along a B-spline curve and stitching the extruded volume onto both surfaces. This
enables, for example, closing a loop in a 1D component-based mesh.

### Element Jacobian damper improvements

> The [ElementJacobianDamper.md] now performs backtracking, and an incorrect MPI reduction in the

Reviewer: ```suggestion
The [ElementJacobianDamper.md] can now perform backtracking to find a working damping factor, when the full update causes excessive deformation of elements make libMesh throw an exception, and an incorrect MPI reduction in the
```

damper was fixed.

### Per-block custom quadrature types

The quadrature setup now accepts a per-block `custom_types` parameter, so different element types can
request different custom quadrature rules on different subdomains. A `paramError` is raised on a
size mismatch.

### Distributed 3D mortar segment meshes

The 3D mortar segment mesh can now be written out in parallel, using per-rank prefix-scan IDs for the
segment mesh nodes and elements. A new `MortarSegmentMeshReporter` was added to output the mortar
segment mesh.

### Dirac kernel improvements

Dirac kernels no longer spuriously report a "point not found" warning in parallel simply because the

> point was not found on the local process, and the element selected when multiple candidates exist is

Reviewer: ```suggestion
point was not found on a MPI rank (it is usually being found on another rank when the mesh distributed), and the element selected when multiple candidates exist is
```

now resolved deterministically. The point lookup was also made faster by removing two map lookups per
Dirac kernel on distributed meshes, and the error message now provides context on the kernel involved.

### Miscellaneous framework improvements

- The Exodus output gained a `max_output_name_length` option, defaulting to the historical 32-character
  limit, and mesh generators no longer truncate Exodus names.
- Mesh generators are now more robust when handling unprepared mesh inputs, with additional caching of
  element subdomain data and clearer comments on which operations invalidate a cached point locator.
- The `Console` output behavior was rationalized, the checkpoint `execute_on` default is now encoded as
  user-set, and a dead `override_default` parameter was removed.
- The automated wall-time checkpoint creation is now easier to disable, since users were unhappy about

>   receiving checkpoints when `checkpoint=false`.

Reviewer: ```suggestion
  still creating walltime-based checkpoints when `checkpoint=false`.
```

- `dataStore`/`dataLoad` support was added for `RestartableDataReporter::Value`.

> - A solution invalidity synchronization is now performed on the final solve.

Reviewer: ```suggestion
- A solution invalidity synchronization is now performed on the final execution step.
```

> - Diagnostics were improved for `UserObject`s and `VectorPostprocessor`s whose getters are queried too

Reviewer: ```suggestion
- `UserObject`s and `VectorPostprocessor`s whose getters are queried too
```

>   early in the setup.

Reviewer: ```suggestion
  early in the setup will now provide a clear error message to application developers
```

- Array variables can now be reinitialized on all the nodal degrees of freedom of multiple nodes at once.
- MooseDocs page breadcrumbs now link to the relevant system syntax page.

> - The deprecated `ADParsedFunction` alias and the `value`, `vars`, `vals`, and `value_{x,y,z}` parameters

Reviewer: ```suggestion
- The deprecated `ADParsedFunction` alias and the `value`, `vars`, `vals`, and `value_{x,y,z}` (for the `ParsedVectorFunction`) parameters
```

Author: Addressed another way in acf4a95.

## MOOSE Bug Fixes

- The `_pattern_pitch` data member of [PatternedHexMeshGenerator.md] is now default-initialized to zero
  rather than left uninitialized.
- A one-component array variable (declared with `array = true`) now correctly receives array-type
  initial conditions; previously `AddVariableAction` classified it as a standard variable based solely
  on its component count.
- Element dampers now skip elements on which their variable has no degrees of freedom (for example a
  block-restricted variable), instead of attempting to evaluate the damper there. A check that the
  element damper variable is scalar was also added.
- Closed parsed curve generation no longer creates a zero-distance node interval where the curve closes
  on itself; the spacing now uses the distance to a curve midpoint for the closing segment.

## MOOSE Modules Changes

### Stochastic Tools module

The Latin hypercube sampling was reworked to use a random permutation, encapsulated in the new
`MooseRandomPerturbation` utility, and the `MooseServer` was extended to produce probability density
function (PDF) and cumulative distribution function (CDF) plots for distributions.

### Subchannel module

> - The repeated laminar/transitional/turbulent Nusselt-number blending used by the heat transfer

Reviewer: looks like it missed commits that were in PRs but older than this month?

Author: Has the missing content been added?

Reviewer: No I have not fixed that

@kyriv1980 if you have time

Reviewer 3: The description seems accurate. The referenced method (blendTurbulentNusseltNumber) has been moved into the base class.

Reviewer: it's missing the point, the mixing closure system got completely reworked
that blending thing is a trivial detail

Author: Content expanded on in 7305113

The [HeliumFluidProperties.md] gained AD support for computing the specific enthalpy `h` from pressure
and temperature, along with its derivatives.

### Navier Stokes module

The weakly-compressible Navier-Stokes Physics now automatically set the time derivatives of constant
variables to zero, and a warning that was unfairly raised when using `system_names` and user objects
for the pressure constraint was silenced.

### Porous Flow module

The `PorousFlowUnsaturated` action now forwards the user-specified residual saturation `s_res` to the
van Genuchten capillary pressure user object it creates. Previously that residual saturation was
dropped, so the capillary pressure was always built with a residual saturation of zero.

### Phase Field module

The [InterfaceNormalCurvatures.md] was added to compute the normal and mean curvatures of an interface.

### Reactor module

The Constructive Solid Geometry (CSG) support was improved: a check for orphaned cells was added to
`CSGBase`, void cells are now removed in pins and assemblies with the existing regions extended out to
infinity, and axis rotations are now applied to assemblies built from x-orientation lattices (the code

> previously assumed a y-orientation lattice).

Reviewer: @kkiesling or @shikhar413 this good enough?

Reviewer 4: I think this is better off talking more generally about how AssemblyMeshGenerator now supports CSG generation, something like

“The Constructive Solid Geometry support has been extended to `AssemblyMeshGenerator`. Any mesh input that relies on `PinMeshGenerator` and `AssemblyMeshGenerator` can be converted to a CSG equivalent definition using `--csg-only`”

Reviewer: thx. edited

  and magnitude of vector variables. These coefficients can be written out to a [MFEMParaViewDataCollection.md].
  ([idaholab/moose#32973](https://github.com/idaholab/moose/pull/32973))
- The [MFEMNDtoRTAux.md] auxiliary kernel was added to copy the degrees of freedom, in 2D, from a source
  variable on an $H(\mathrm{curl})$ Nédélec space into a target variable on an $H(\mathrm{div})$
  Raviart-Thomas space. In 2D, the two bases are related by a 90-degree rotation, which is useful in, e.g.
  magnetostatic postprocessing for obtaining the magnetic flux density from a rotated vector potential gradient.
  ([idaholab/moose#32923](https://github.com/idaholab/moose/pull/32923))
- Several cleanups were made to the MFEM backend code and its documentation.
  ([idaholab/moose#32781](https://github.com/idaholab/moose/pull/32781),
  [idaholab/moose#32884](https://github.com/idaholab/moose/pull/32884))

### Kokkos Backend Improvements

- Automatic differentiation (AD) support was added. Currently supported objects are `ADKernel`, `ADIntegratedBC`, and `ADNodalBC`. See [this page](syntax/KokkosKernels/index.md#kokkos_ad_kernel) for more details.

> - The residual object registration macros were unified. All residual objects (kernels, nodal kernels, and boundary conditions) should be now registered with `registerKokkosResidualObject` (`registerKokkosADResidualObject` for AD versions). Aliases like `registerKokkosKernel` are no longer supported.

Author: ```suggestion
- The residual object registration macros were unified. All residual objects (kernels, nodal kernels, and boundary conditions) should now be registered with `registerKokkosResidualObject` (`registerKokkosADResidualObject` for AD versions). Aliases like `registerKokkosKernel` are no longer supported.
```

`bin_number` parameters.

### Manifold-based subdomain assignment

The [ManifoldSubdomainGenerator.md] was added to assign a subdomain ID based on whether an element's
vertex-average point lies inside a closed (watertight) surface mesh, supplied through the mesh
generator pipeline. Beyond tagging elements inside or outside an arbitrary surface, it can be used to
"voxelize" a geometry defined by a surface mesh by removing elements of a structured grid that fall
inside or outside the manifold, which is useful when exactly fitting to the boundary is not required.

### Advanced extrusion along a curve and radial growth

The [AdvancedExtruderGenerator.md] can now extrude a mesh following a line mesh / curve rather than a
fixed direction, with the extrusion direction computed locally at every curve node. The starting and
ending directions can optionally be imposed. In addition, the radial extent of the extruded mesh can
be progressively expanded or diminished throughout the extrusion by setting the `end_radial_extent`
parameter.

### Polygon mesh extrusion in parallel

The conversion of polygon meshes to tetrahedra and the extrusion of polygon meshes now work with

> distributed meshes, with unique ID and midnode numbering made consistent in parallel. The optional

Author: ```suggestion
distributed meshes, with unique ID and mid-node numbering made consistent in parallel. The optional
```

PNG output was also made more robust and additional parameters were added to better select the area to plot.

### Junction component

The [ComponentJunction.md] action component was added to create a contiguous junction between two
components, either by stitching meshes that are already in contact or by extruding one boundary
towards the other along a B-spline curve and stitching the extruded volume onto both surfaces. This
enables, for example, closing a loop in a 1D component-based mesh.

### Element Jacobian damper improvements

The [ElementJacobianDamper.md] can now perform backtracking to find a working damping factor, when the full update causes excessive deformation of elements make libMesh throw an exception, and an incorrect MPI reduction in the

> damper was fixed.

Author: ```suggestion
The [ElementJacobianDamper.md] can now perform backtracking to find a working damping factor, when the full update causes excessive deformation of elements and makes libMesh throw an exception, and an incorrect MPI reduction in the
damper was fixed.
```

### Navier Stokes module

The weakly-compressible Navier-Stokes Physics now automatically set the time derivatives of constant
variables to zero, and a warning that was unfairly raised when using `system_names` and user objects
for the pressure constraint was silenced.

### Porous Flow module

The `PorousFlowUnsaturated` action now forwards the user-specified residual saturation `s_res` to the
van Genuchten capillary pressure user object it creates. Previously that residual saturation was
dropped, so the capillary pressure was always built with a residual saturation of zero.

### Phase Field module

The [InterfaceNormalCurvatures.md] was added to compute the normal and mean curvatures of an interface.
To allow compilation in `opt` mode with gcc 15 and 16, which crash with an internal compiler error
(segmentation fault) on a number of class constructors, we are now explicitly preventing inlining of
[EBTerm](ExpressionBuilder.md)'s default constructor.

### Reactor module

The Constructive Solid Geometry (CSG) support has been extended to the [AssemblyMeshGenerator.md].
Any mesh input that relies on [PinMeshGenerator.md] and [AssemblyMeshGenerator.md] can now be

> converted to an equivalent CSG definition by passing `--csg-only`. The axial extent of these regions is currently infinite, the CSG generation is not currently implemented for extruded 3D assemblies with multiple axial definitions. As part of this work, the system now checks for orphaned constructs (cells, universes) in `CSGBase`.

Author: ```suggestion
converted to an equivalent CSG definition by passing `--csg-only`. The axial extent of these regions is currently infinite; the CSG generation is not currently implemented for extruded 3D assemblies with multiple axial definitions. As part of this work, the system now checks for orphaned constructs (cells, universes) in `CSGBase`.
```

- Added Netgen control for tetrahedralization multithreading; set Netgen thread count to match libMesh n_threads
- More robust build system behavior when rebuilding after Netgen updates
- More assertions in AnalyticFunction

> - Fix for compiler warning in libMesh from macOS 26.4

Author: ```suggestion
- Fix for compiler warning in libMesh from MacOS 26.4
```

Reviewer 5: ❌ nah. :-P

Author: Bah. You're right. :P Thanks!

> - Fix for compiler warning in libMesh from MacOS 26.4

Author: ```suggestion
- Fix for compiler warning in libMesh from macOS 26.4
```
